packages/@aws-cdk/aws-ec2/lib/network-acl.ts

```diff
@@ -0,0 +1,411 @@
+import { CfnNetworkAcl, CfnNetworkAclEntry, CfnSubnetNetworkAclAssociation  } from './ec2.generated';
+import { IVpc } from './vpc';
+
+export interface INetworkACL extends IResource {
+  /**
+   * ID for the current Network ACL
+   * @attribute
+   */
+  readonly networkAclId: string;
+}
+
+/**
+ * A NetworkAclBase that is not created in this template
+ */
+abstract class NetworkAclBase extends Resource implements INetworkACL {
+public abstract readonly networkAclId: string;
+constructor(scope: Construct, id: string, props?: ResourceProps) {
+    super(scope, id, props);
+
+  }
+
+  public get uniqueId() {
+    return this.node.uniqueId;
+  }
+}
+
+export interface NetworkAclProps {
+  /**
+   * The name of the NetworkAcl.
+   *
+   * It is not recommended to use an explicit name.
+   *
+   * @default If you don't specify a networkAclName, AWS CloudFormation generates a
+   * unique physical ID and uses that ID for the group name.
+   */
+  readonly networkAclName?: string;
+  /**
+   * The VPC in which to create the NetworkACL.
+   */
+  readonly vpc: IVpc;
+}
+
+export class NetworkAcl extends NetworkAclBase {
+  /**
+   * Import an existing NetworkAcl into this app.
+   */
+  public static fromNetworkAclId(scope: Construct, id: string, networkAclId: string): INetworkACL {
+  class Import extends NetworkAclBase {
+    public networkAclId = networkAclId;
+    }
+
+  return new Import(scope, id);
+  }
+  /**
+   * The ID of the NetworkACL
+   *
+   * @attribute
+   */
+  public readonly networkAclId: string;
+
+  /**
+   * The VPC ID for this NetworkACL
+   *
+   * @attribute
+   */
+  public readonly networkAclVpcId: string;
+
+  private readonly networkACL: CfnNetworkAcl;
+
+  constructor(scope: Construct, id: string, props: NetworkAclProps) {
+    super(scope, id, {
+      physicalName: props.networkAclName
+    });
+    this.networkACL = new CfnNetworkAcl(this, 'Resource', {
+      vpcId: props.vpc.vpcId,
+    });
+
+    this.networkAclId = this.networkACL.ref;
+    this.networkAclVpcId = this.networkACL.vpcId;
+  }
+}
+
+export interface Icmp {
+  /**
+   * The Internet Control Message Protocol (ICMP) code. You can use -1 to specify all ICMP
+   * codes for the given ICMP type. Requirement is conditional: Required if you
+   * specify 1 (ICMP) for the protocol parameter.
+   */
+  readonly code?: number
+  /**
+   * The Internet Control Message Protocol (ICMP) type. You can use -1 to specify all ICMP types.
+   * Conditional requirement: Required if you specify 1 (ICMP) for the CreateNetworkAclEntry protocol parameter.
+   */
+  readonly type?: number
+}
+
+export interface PortRange {
+  /**
+   * The first port in the range. Required if you specify 6 (TCP) or 17 (UDP) for the protocol parameter.
+   */
+  readonly from?: number
+  /**
+   * The last port in the range. Required if you specify 6 (TCP) or 17 (UDP) for the protocol parameter.
+   */
+  readonly to?: number
+}
+
+export interface INetworkAclEntry extends IResource {
```

Comment:
In principle `INetworkAclEntry` only needs `networkAclEntryName: string` as a property, not all these configuration properties. 

Do you need them for some other purpose?

Reply:
removed the extra properties